pkg/virt-controller/watch/clone/clone_base.go

```diff
@@ -144,13 +144,19 @@ func (ctrl *VMCloneController) handleSnapshot(obj interface{}) {
 		return
 	}
 
-	keys, err := ctrl.vmCloneInformer.GetIndexer().IndexKeys("snapshotSource", snapshotKey)
+	snapshotSourceKeys, err := ctrl.vmCloneInformer.GetIndexer().IndexKeys("snapshotSource", snapshotKey)
 	if err != nil {
-		log.Log.Object(snapshot).Reason(err).Error("cannot get clone keys from snapshotSource indexer")
+		log.Log.Object(snapshot).Reason(err).Error("cannot get clone snapshotSourceKeys from snapshotSource indexer")
 		return
 	}
 
-	for _, key := range keys {
+	snapshotWaitingKeys, err := ctrl.vmCloneInformer.GetIndexer().IndexKeys(string(clonev1alpha1.SnapshotInProgress), snapshotKey)
```

Comment:
hi， could you please inform me why we need  to get snapshotWaitingKeys and restoreWaitingKeys? In my opinion, all these snapshots and restores are owned by a clone so we have add these snapshots and restores to the Queue in 
```
if ownedByClone, key := isOwnedByClone(snapshot); ownedByClone {
		ctrl.vmCloneQueue.AddRateLimited(key)
	}
```
and
```
if ownedByClone, key := isOwnedByClone(restore); ownedByClone {
		ctrl.vmCloneQueue.AddRateLimited(key)
	}
```

Reply:
Hey @Longchuanzheng! Thanks for reviewing.

> n my opinion, all these snapshots and restores are owned by a clone

This is not always correct. It is correct when the clone source is a VM, then a snapshot is being created by the clone controller. However, when the source is a snapshot (that already exists and created by someone else), the snapshot is not owned by any clone object.

Hope this helps clarifying!